cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -197,37 +261,37 @@ case class CommandLineTool private(
         case other => throw new NotImplementedError(s"command input parameters such as $other are not yet supported")
       }.toList
 
-      def stringOrExpressionToString(soe: Option[StringOrExpression]): Option[String] = soe flatMap {
-        case StringOrExpression.String(str) => Some(str)
-        case StringOrExpression.Expression(_) => None // ... for now!
-      }
-
-      // The try will succeed if this is a task within a step. If it's a standalone file, the ID will be the file,
-      // so the filename is the fallback.
-      def taskName = Try(FullyQualifiedName(id).id).getOrElse(Paths.get(id).getFileName.toString)
-
-      val adHocFileCreations: Set[WomExpression] = (for {
-        requirements <- requirements.getOrElse(Array.empty[Requirement])
-        initialWorkDirRequirement <- requirements.select[InitialWorkDirRequirement].toArray
-        listing <- initialWorkDirRequirement.listings
-      } yield InitialWorkDirFileGeneratorExpression(listing, expressionLib)).toSet[WomExpression]
-
-      CallableTaskDefinition(
-        taskName,
-        buildCommandTemplate(expressionLib),
-        runtimeAttributes,
-        meta,
-        parameterMeta,
-        outputs,
-        inputDefinitions,
-        // TODO: This doesn't work in all cases and it feels clunky anyway - find a way to sort that out
-        prefixSeparator = "#",
-        commandPartSeparator = " ",
-        stdoutRedirection = stringOrExpressionToString(stdout),
-        stderrRedirection = stringOrExpressionToString(stderr),
-        adHocFileCreation = adHocFileCreations
-      )
+    def stringOrExpressionToString(soe: Option[StringOrExpression]): Option[String] = soe flatMap {
+      case StringOrExpression.String(str) => Some(str)
+      case StringOrExpression.Expression(_) => None // ... for now!
     }
+
```

Comment:
ToL is this section different?  Not sure why Github marked this whole area in the diff as it seems identical save for an extra argument to `CallableTaskDefinition`

Reply:
It was indented an extra level incorrectly before. I used to have this PR chopped up into two commits where these changes were separate, but I've had to rewrite this twice now due to rebases so unfortunately that nicety was lost.